lib/matplotlib/tests/test_ticker.py

```diff
@@ -524,26 +524,93 @@ def test_basic(self, format, input, expected):
-        instances with and without units. Cases focus on when no SI
-        prefix is present, for values in [1, 1000).
+        Test the formatting of EngFormatter for various values of the 'places'
+        argument, in several cases:
+            0.  without a unit symbol but with a (default) space separator;
+            1.  with both a unit symbol and a (default) space separator;
+            2x. with both a unit symbol and some non default separators;
+            3x. without a unit symbol but with some non default separators.
         """
-        fmt = mticker.EngFormatter(unit)
-        assert fmt(input) == expected
+
+        UNIT = 's'  # seconds
+        DIGITS = '0123456789'  # %timeit showed 10-20% faster search than set
+
+        # Case 0: unit='' (default) and sep=' ' (default).
+        # 'expected' already corresponds to this reference case.
+        exp_outputs = (_s for _s in expected)  # simple copy of 'expected'
+        formatters = (
+            mticker.EngFormatter(),  # places=None (default)
+            mticker.EngFormatter(places=0),
+            mticker.EngFormatter(places=2)
+        )
+        for _formatter, _exp_output in zip(formatters, exp_outputs):
+            assert _formatter(input) == _exp_output
+
+        # Case 1: unit=UNIT and sep=' ' (default).
+        # Append a unit symbol to the reference case.
+        # Beware of the values in [1, 1000), where there is no prefix!
+        exp_outputs = (_s + " " + UNIT if _s[-1] in DIGITS  # case w/o prefix
+                       else _s + UNIT for _s in expected)
+        formatters = (
+            mticker.EngFormatter(unit=UNIT),  # places=None (default)
+            mticker.EngFormatter(unit=UNIT, places=0),
+            mticker.EngFormatter(unit=UNIT, places=2)
+        )
+        for _formatter, _exp_output in zip(formatters, exp_outputs):
+            assert _formatter(input) == _exp_output
+
+        # Test several non default separators: no separator, a narrow
+        # no-break space (unicode character) and an extravagant string.
+        for _sep in ("", "\u202f", "@_@"):
+            # Case 2x: unit=UNIT and sep=_sep.
+            # Remove the space separator from the reference case.
+            exp_outputs = (_s + _sep + UNIT if _s[-1] in DIGITS  # no prefix
+                           else _s.replace(" ", _sep) + UNIT
+                           for _s in expected)
+            formatters = (
+                mticker.EngFormatter(unit=UNIT, sep=_sep),  # places=None
+                mticker.EngFormatter(unit=UNIT, places=0, sep=_sep),
+                mticker.EngFormatter(unit=UNIT, places=2, sep=_sep)
+            )
+            for _formatter, _exp_output in zip(formatters, exp_outputs):
+                assert _formatter(input) == _exp_output
+
+            # Case 3x: unit='' (default) and sep=_sep.
+            # Remove the space separator from the reference case and append
+            # a unit symbol to it.
```

Comment:
Yes, you are right. The cases 2. and 3. were switched during a previous commit, and I failed at (completely) switching the relevant comments… I will take care of it.